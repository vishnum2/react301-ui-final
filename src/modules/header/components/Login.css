.admin-wrapper{
    background-color: #282c34;
    /* min-height: 100vh;
    min-width: 100vw; */
    /* height: 90vh; */
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    /* font-size: calc(10px + 3vmin); */
    color: white;
}

.inner-jumbotron{
    border-radius: "10px";
    background-color: "grey";
    height: "50vh";
    width: "40vw";
    display: "flex";
    flex-direction: "column";
    justify-content: "Center";
    padding: "2vh"
}

.admin-form-elements{
    /* flex: 1; */
    margin: 2vh !important;
    border: white;
    /* width: 20vw; */
}

.admin-google-login-button{
    background-color: #61dafb;
    padding: 2vh;
    border-radius: 5px;
    cursor: pointer;
}

.admin-google-login-button:hover{
   opacity: 0.8;
}

/* .MuiFormLabel-root{
    color: white !important;
} */