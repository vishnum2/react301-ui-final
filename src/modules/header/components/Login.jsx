import React, {useState} from 'react';
import axiosWrapper from "../../../apis/axiosCreate";
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import { login, createUser } from "../state/actions";
import Dialog from '@material-ui/core/Dialog';
import { GoogleLogin } from 'react-google-login';
import HighlightOffIcon from '@material-ui/icons/HighlightOff';
import {adminEmail} from '../../../config/constants';

// import {
//     withStyles
//   } from '@material-ui/core/styles';
import './Login.css';

export const Login = (props) => {
    // const CustomisedTextField = withStyles({
    //     root: {
    //         '& .MuiFormLabel-root':{
    //             color: 'white'
    //         }
    //     }
    // })(TextField);

    const [formUserName, setFormUserName] = useState('');
    const [formPassword, setFormPassword] = useState('');

    const handleChange = (ev) => {
        // console.log(ev.target)
        if(ev.target.id === 'userName') {
            // console.log(ev.target.value);
            setFormUserName(ev.target.value);
        } else {
            // console.log("password");
            setFormPassword(ev.target.value);
        }
    }

    const responseGoogle = async (response) => {
        console.log(response);
        const isAlreadyUser = await axiosWrapper.get(`/users?email=${response.profileObj.email}`);
        // console.log(isAlreadyUser);
        if(isAlreadyUser.data.length === 0) {
            const temp = {
                            "id": response.profileObj.googleId,
                            "email": response.profileObj.email,
                            "first_name": response.profileObj.givenName,
                            "last_name": response.profileObj.familyName,
                            "picture": response.profileObj.imageUrl,
                            "balance_points": 0,
                            "wishlist": [],
                            "cards_gifted": [],
                            "cards_received": [] 
                         }
            props.createUser(temp);
            props.login(temp);
            window.sessionStorage.setItem('user', JSON.stringify(temp));
        } else {
            props.login(isAlreadyUser.data[0]);
            window.sessionStorage.setItem('user', JSON.stringify(isAlreadyUser.data[0]));
        }
        props.handleDialogStatus(false);
    }
  
    const login = async() => {
        console.log(formUserName);
        if(formUserName === adminEmail[0]) {
          const temp = {
            "id": 1,
            "email": formUserName,
            "first_name": 'admin',
            "last_name": 'admin',
            "picture": '',
            "balance_points": 0,
            "wishlist": [],
            "cards_gifted": [],
            "cards_received": [] 
         }
         console.log(temp);
         props.login(temp);
          window.sessionStorage.setItem('user', JSON.stringify(temp));
          window.sessionStorage.setItem('usertype', 'admin');
        } else {
          const isaUser = await axiosWrapper.get(`/users?email=${formUserName}`);
          console.log(isaUser);
          if(isaUser.data.length !== 0 && isaUser.data[0].password === 'password') {
            console.log(isaUser.data[0]);
              props.login(isaUser.data[0]);
              window.sessionStorage.setItem('user', JSON.stringify(isaUser.data[0]));
          } else {
              console.log('not a registered user or credentials wrong');
          }
        }
        props.handleDialogStatus(false);
    }

    return (
        <>
        <Dialog open={props.open} aria-labelledby="form-dialog-title" fullWidth>
            {/* <DialogTitle id="form-dialog-title">Edit Services</DialogTitle> */}
            <div>
            <HighlightOffIcon testData='offIcon' onClick={() => props.handleDialogStatus(false)} style={{float: "right", cursor: 'pointer'}}/>
            </div>
            <div className="admin-wrapper">
                <h1>
                    Login
                </h1>
                {/* <CustomisedTextField
                  className="admin-form-elements"
                  id="userName"
                  hintText="Text"
                  label="User Name"
                  value={formUserName}
                  variant="outlined"
                  uniqueIdentifier="userName"
                  onChange={handleChange}
                />  */}
                <TextField
                  className="admin-form-elements"
                  id="userName"
                  hintText="Text"
                  label="User Name"
                  value={formUserName}
                  variant="outlined"
                  uniqueIdentifier="userName"
                  onChange={handleChange}
                /> 


                <TextField
                  className="admin-form-elements"
                  id="password"
                  hintText="password"
                  label="Password"
                  value={formPassword}
                  variant="outlined"
                  type="password"
                  onChange={handleChange}
                />

                <Button testData='login-button' className="admin-form-elements" variant="contained" color="primary" onClick={login}>
                  Login
                </Button>
                <h4>Or</h4>
                <div>
                {/* <a style={{textDecoration: "none", color: "white"}} href="http://localhost:8080/api/login/google"> Login using Google</a> */}
                <GoogleLogin
                  clientId="430181379102-p693ifhk3pnuvhhia1f9e701kp3sphoc.apps.googleusercontent.com"
                  buttonText="Login Using Google"
                  onSuccess={responseGoogle}
                  onFailure={responseGoogle}
                  cookiePolicy={'single_host_origin'}
                />
                </div>
                <p style= {{color: "white"}}>Please use external network for oauth login if Mindtree network blocks it</p>
                </div>
        </Dialog>
        </>
    )
}

export const mapStateToProps = state => {
    return {};
  };
  
 export const mapDispatchToProps = dispatch => {
    return bindActionCreators({ login, createUser }, dispatch);
  };
  

export default connect(mapStateToProps, mapDispatchToProps)(Login);