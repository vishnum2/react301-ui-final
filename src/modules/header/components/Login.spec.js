import React from 'react';
import {shallow} from '../../../enzyme';
import {Login} from './Login';
import axiosWrapper from '../../../apis/axiosCreate';
import {mapStateToProps, mapDispatchToProps} from './Login';

describe('snapshot of component', () => {
    it('should match snapshot', () => {
        const dialogStatusMock = jest.fn() 
        const wrapper = shallow(<Login  handleDialogStatus={dialogStatusMock} open={true}/>);
        expect(wrapper).toMatchSnapshot();
        wrapper.find({testData: 'offIcon'}).simulate('click')
        expect(dialogStatusMock).toHaveBeenCalled();
    })
});

describe('test simulate and check functions', () => {
    let mockLogin;
    let dialogStatusMock;

    beforeEach(() => {
        mockLogin = jest.fn().mockImplementationOnce(() => Promise.resolve({}));
        dialogStatusMock = jest.fn() 
    })
    
    it('should simulate admin login', () => {
        const props = {};
        const wrapper = shallow(<Login formUserName='new' handleDialogStatus={dialogStatusMock} login={mockLogin} {...props} open={true}/>);
        wrapper.find({hintText: 'Text'}).simulate('change', {
            target: {
                id: 'userName',
                value: 'yoyogiftg2@gmail.com'
            }
        });
        wrapper.find({hintText: 'password'}).simulate('change', {
            target: {
                id: 'password',
                value: 'password'
            }
        });
        wrapper.find({testData: 'login-button'}).simulate('click');
        const response = { id: 1,
            email: 'yoyogiftg2@gmail.com',
            first_name: 'admin',
            last_name: 'admin',
            picture: '',
            balance_points: 0,
            wishlist: [],
            cards_gifted: [],
            cards_received: [] };
            expect(mockLogin).toHaveBeenCalledWith(response);
        });
        it('should simulate error user', () => {
            
            axiosWrapper.get = jest.fn().mockImplementationOnce(() => Promise.resolve({data: []}))
            const wrapper = shallow(<Login handleDialogStatus={dialogStatusMock} open={true} />);
            wrapper.find({hintText: 'Text'}).simulate('change', {
                target: {
                    id: 'userName',
                    value: 'vishnu@gmail.com'
                }
            });
            wrapper.find({hintText: 'password'}).simulate('change', {
                target: {
                    id: 'password',
                    value: 'password'
                }
            });
            wrapper.find({testData: 'login-button'}).simulate('click');
     });

     it('should simulate user login', async () => {
        axiosWrapper.get = jest.fn().mockImplementationOnce(() => Promise.resolve({data: [{password: 'password'}]}));
        const wrapper = shallow(<Login login={mockLogin} handleDialogStatus={dialogStatusMock}  open={true} />);
        wrapper.find({hintText: 'Text'}).simulate('change', {
            target: {
                id: 'userName',
                value: 'vishnu@gmail.com'
            }
        });
        wrapper.find({hintText: 'password'}).simulate('change', {
            target: {
                id: 'password',
                value: 'password'
            }
        });
        wrapper.find({testData: 'login-button'}).simulate('click');
        await expect(axiosWrapper.get).toHaveBeenCalled();
     });

     it('should google login', async () => {
        axiosWrapper.get = jest.fn().mockImplementationOnce(() => Promise.resolve({data: []}));
        const userCreateMock = jest.fn().mockImplementationOnce(() => Promise.resolve());
        const wrapper = shallow(<Login createUser={userCreateMock} handleDialogStatus={dialogStatusMock} login={mockLogin}  open={true} />);
        wrapper.find({buttonText: "Login Using Google"}).prop('onSuccess')({profileObj: {
            googleId: '122',
            email: 'vishnu',
            givenName: 'vishnu',
            familyName: 'priyan',
            imageUrl: 'nothing'
        }});
        await expect(axiosWrapper.get).toHaveBeenCalled();
     });

     it('should detect already stored user', () => {
        axiosWrapper.get = jest.fn().mockImplementationOnce(() => Promise.resolve({data: [{data: 'data'}]}));
        const wrapper = shallow(<Login handleDialogStatus={dialogStatusMock} login={mockLogin}  open={true} />);
        wrapper.find({buttonText: "Login Using Google"}).prop('onSuccess')({profileObj: {
            googleId: '122',
            email: 'vishnu',
            givenName: 'vishnu',
            familyName: 'priyan',
            imageUrl: 'nothing'
        }});
        // expect(mockLogin).toHaveBeenCalledWith({data: 'data'})
     })
});

describe('check state', () => {
    it('should check mapStateToProps', () => {
        const state = {}
        mapDispatchToProps();
        expect(mapStateToProps(state)).toEqual({})
    })
})