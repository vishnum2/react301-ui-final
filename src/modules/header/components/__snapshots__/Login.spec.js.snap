// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of component should match snapshot 1`] = `ShallowWrapper {}`;
